### DISCUSSION TITLE
Advice on how to implement custom shortcuts to open and switch between commonly used apps

### DISCUSSION THREAD

**Discussion by Aidan-McGuire** [manna-harbour/miryoku#367] [Q&A] [answered]
Hello! I've been using Miriyoku on a Corne for about a year, I really enjoy it, thank you for all your hard work on this! I'd like to start using custom shortcuts to open and switch between my frequently used apps instead of cmd + tab + tab + tab...

I'd like some advice on how to best implement this behavior. My first thought was to create custom shortcuts for each number + all modifiers together, to avoid conflicts with existing shortcuts across various apps (i.e. cmd + alt + ctrl + shft + 1 would switch to my browser). It looks like custom shortcuts can be assigned this with with something like [Snap](https://apps.apple.com/us/app/snap/id418073146?mt=12) or [Raycast](https://www.raycast.com/). I'd love to hear y'alls thoughts on how to implement something like this. Thanks

Bonus question, is it possible to bind multiple modifiers + a character to a single key press?

**Comment by manna-harbour** [comment, marked as answer]
Miryoku is designed so that any combination of mods and another key can be performed without any contortions and with minimum movement. For app switching, you can usually switch using the OS default shortcuts with a mod and number, letter, or arrows. If you need to add custom shortcuts I'd recommend Super + number or letter. If you can't do it with a single mod you can hold many mods simultaneously, or replace AltGr (below Alt on both sides) with multiple mods using https://github.com/manna-harbour/miryoku/discussions/85.

Multiple mods can be nested around a single key press e.g. https://zmk.dev/docs/keymaps/modifiers#modifier-functions, https://docs.qmk.fm/feature_advanced_keycodes#modifier-keys.

**Comment by gihrig** [comment]
I manage app switching with Alfred, Snap or Raycast work about the same. Alfred learns most commonly used abbreviations. So I have Cmd-Space for Alfred then 'f' for firefox, 'm' for mail, 'b' for BB Edit, 'br' for Brave, etc., etc.
